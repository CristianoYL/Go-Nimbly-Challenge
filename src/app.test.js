import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './app';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('App contents not modified', () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
