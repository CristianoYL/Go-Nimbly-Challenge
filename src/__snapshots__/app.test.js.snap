// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App contents not modified 1`] = `
<div
  className="full-height container-fluid"
>
  <div
    className="full-height row"
  >
    <div
      className="full-height no-padding col-sm-4"
    >
      <div
        className="user-panel"
      >
        <div
          className="text-field-container"
        >
          <div
            className="message-text"
          >
            <h2>
              Enter your math expression and see the magic!
            </h2>
          </div>
          <div
            className="menu-container full-width"
          >
            <label
              htmlFor="operation-select"
              target="_select"
            >
              select a math operation
              <select
                id="operation-select"
                onChange={[Function]}
                value="default"
              >
                <option
                  disabled={true}
                  value="default"
                >
                  -- select an operation --
                </option>
                <option
                  value="simplify"
                >
                  Simplify
                </option>
                <option
                  value="factor"
                >
                  Factor
                </option>
                <option
                  value="derive"
                >
                  Derive
                </option>
                <option
                  value="integrate"
                >
                  Integrate
                </option>
                <option
                  value="zeroes"
                >
                  Find 0s
                </option>
              </select>
            </label>
          </div>
          <div
            className="text-field full-width"
          >
            <label
              htmlFor="your math expression*"
              target="_input"
            >
              your math expression*
              <input
                id="your math expression*"
                onChange={[Function]}
                placeholder="your math expression*"
                value=""
              />
            </label>
          </div>
          <button
            className="btn-submit"
            onClick={[Function]}
          >
            Get Result
          </button>
        </div>
      </div>
    </div>
    <div
      className="full-height no-padding col-sm-8"
    >
      <div
        className="display-panel"
      >
        <div
          className="display-text"
        >
          <div
            className="app-header"
          >
            <img
              alt="logo"
              className="app-logo"
              src="logo.png"
            />
            <p
              className="header-text"
            >
              Choose an operation and submit your math expression, the result will be displayed below
            </p>
          </div>
          <h3>
            Here is your math result
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;
